test/io/jade/readByte-throws.chpl

```diff
@@ -0,0 +1,19 @@
+use IO;
+
+var f = openTempFile();
+
+{
+  var w = f.writer();
+  w.writeByte(0x11);
+}
+
+{
+  var r = f.reader();
+  r.readByte(); // should not throw
+  try {
+    r.readByte(); // should throw
+    assert(false);
```

Comment:
I don't generally work in this area so my opinion on it is not too important (i.e. you can disregard it after considering), but I personally generally prefer to have testing based on "does the .good file match with these prints" rather than testing "does the .chpl file compile with these asserts", at least for the happy path.

For example, in this case imagine the good file could contain lines like (off the top of my head) "valid readbyte didn't throw" and "invalid readbyte threw" corresponding to `writeln`s in place of your asserts. IMO that provides a small improvement in clarity for future readers, who now don't necessarily even need to read the `.chpl` of the test to have some understanding of what is being tested. Additionally if we change logic in the future that makes it more obvious to me what needs to change.